### ISSUE TITLE
Why the loss value changes greatly between epochs?

### ISSUE THREAD

**Issue by LokLu** [keras-team/keras#7648]
I run the mnist_cnn example under the 'examples' folder. Below is the log informaiton.
I wonder why the loss value changed greatly between epochs.
Is there any special calculation between epochs?
What's the difference between the batch by batch update and epoch by epoch update?

56448/60000 [===========================>..] - ETA: 0s - loss: 0.3547 - acc: 0.8938
56960/60000 [===========================>..] - ETA: 0s - loss: 0.3529 - acc: 0.8942
57344/60000 [===========================>..] - ETA: 0s - loss: 0.3515 - acc: 0.8946
57856/60000 [===========================>..] - ETA: 0s - loss: 0.3497 - acc: 0.8952
58240/60000 [============================>.] - ETA: 0s - loss: 0.3484 - acc: 0.8954
58624/60000 [============================>.] - ETA: 0s - loss: 0.3468 - acc: 0.8960
59136/60000 [============================>.] - ETA: 0s - loss: 0.3449 - acc: 0.8965
59648/60000 [============================>.] - ETA: 0s - loss: 0.3436 - acc: 0.8969
60000/60000 [==============================] - 9s - loss: 0.3423 - acc: 0.8972 - val_loss: 0.0793 - val_acc: 0.9767
Epoch 2/12
  128/60000 [..............................] - ETA: 9s - loss: 0.0927 - acc: 0.9766
  640/60000 [..............................] - ETA: 7s - loss: 0.1146 - acc: 0.9672
 1024/60000 [..............................] - ETA: 7s - loss: 0.1318 - acc: 0.9658
 1408/60000 [..............................] - ETA: 7s - loss: 0.1307 - acc: 0.9645
 1792/60000 [..............................] - ETA: 7s - loss: 0.1323 - acc: 0.9626
 2176/60000 [>.............................] - ETA: 7s - loss: 0.1248 - acc: 0.9637
 2688/60000 [>.............................] - ETA: 7s - loss: 0.1241 - acc: 0.9643

**Comment by ozabluda** [comment]
Loss (0.3423) is average across the whole epoch 1. Loss (0.0927) is the loss of the first minibatch of the epoch 2.
